showcase/gapic-showcase/src/test/java/com/google/showcase/v1beta1/it/ITClientShutdown.java

```diff
@@ -0,0 +1,188 @@
+    // the termination time should only take about 2s (time to receive a response) + time
+    // to close the client. Check that this takes less than 5s (2s request time + 3s
+    // buffer time).
+    long terminationTime = end - start;
+    Truth.assertThat(terminationTime).isLessThan(5000L);
+  }
+
+  // This test is to ensure that the client is able to close + terminate any resources
+  // once a response has been received. Set a max test duration of 15s to ensure that
+  // the test does not continue on forever.
+  @Test(timeout = 15000L)
+  public void testHttpJson_rpcInvokedWithLargeTimeout_closeClientOnceResponseReceived()
+      throws Exception {
+    // Set the maxAttempts to 1 to ensure there are no retries scheduled. The single RPC
+    // invocation should time out in 15s, but the client will receive a response in 2s.
+    // Any outstanding tasks (timeout tasks) should be cancelled once a response has been
+    // received so the client can properly terminate.
+    RetrySettings defaultRetrySettings =
+        RetrySettings.newBuilder()
+            .setInitialRpcTimeout(Duration.ofMillis(15000L))
+            .setMaxRpcTimeout(Duration.ofMillis(15000L))
+            .setTotalTimeout(Duration.ofMillis(15000L))
+            .setMaxAttempts(1)
+            .build();
+    EchoClient httpjsonClient =
+        TestClientInitializer.createHttpJsonEchoClientCustomBlockSettings(
+            defaultRetrySettings, ImmutableSet.of(StatusCode.Code.DEADLINE_EXCEEDED));
+
+    BlockRequest blockRequest =
+        BlockRequest.newBuilder()
+            .setSuccess(BlockResponse.newBuilder().setContent("httpjsonBlockContent_2sDelay"))
+            .setResponseDelay(com.google.protobuf.Duration.newBuilder().setSeconds(2).build())
+            .build();
+
+    long start = System.currentTimeMillis();
+    BlockResponse response = httpjsonClient.block(blockRequest);
+    Truth.assertThat(response.getContent()).isEqualTo("httpjsonBlockContent_2sDelay");
+
+    // Intentionally do not run grpcClient.awaitTermination(...) as this test will
+    // check that everything is properly terminated after close() is called.
+    httpjsonClient.close();
+
+    busyWaitUntilClientTermination(httpjsonClient);
+    long end = System.currentTimeMillis();
+
+    Truth.assertThat(httpjsonClient.isShutdown()).isTrue();
+
+    // Check the termination time. If all the tasks/ resources are closed successfully,
+    // the termination time should only take about 2s (time to receive a response) + time
+    // to close the client. Check that this takes less than 5s (2s request time + 3s
+    // buffer time).
+    long terminationTime = end - start;
+    Truth.assertThat(terminationTime).isLessThan(5000L);
+  }
+
+  // Loop until the client has terminated successfully. For tests that use this,
+  // try to ensure there is a timeout associated, otherwise this may run forever.
+  // Future enhancement: Use awaitility instead of busy waiting
```

Comment:
Could you explain what presents us from this enhancement right now?

Reply:
There is currently nothing actively blocking adding the Awaitility library. I think there are quite a few tests inside showcase and gax that could benefit from including it. I didn't include it in this PR b/c I thought it would make this scope of this PR much bigger.

I can create an issue to introduce Awaitility and using it can be done separately.